src/TheSpace/HarbergerMarket.sol

```diff
@@ -69,120 +84,177 @@ abstract contract HarbergerMarket is Multicall {
      * @dev Create Property contract, setup attached currency contract, setup tax rate
      */
     constructor(
-        string memory propertyName_,
-        string memory propertySymbol_,
-        address currencyAddress_,
-        uint256 taxRate_,
-        uint256 totalSupply_
+        string memory propertyName,
+        string memory propertySymbol,
+        address currencyAddress
     ) {
         // initialize Property contract with current contract as market
-        property = new Property(propertyName_, propertySymbol_, address(this), totalSupply_);
+        property = new Property(propertyName, propertySymbol, address(this), totalSupply);
 
         // initialize currency contract
-        currency = ERC20(currencyAddress_);
+        currency = ERC20(currencyAddress);
 
-        // TODO: tax rate setter
-        totalSupply = totalSupply_;
-        taxRate = taxRate_;
+        // default config
+        taxConfig[ConfigOptions.taxRate] = 10;
+        taxConfig[ConfigOptions.treasuryShare] = 500;
     }
 
     /**
      * @dev Set the current price of an Harberger property with token id.
      *
      * Emits a {Price} event.
      */
-    function setPrice(uint256 tokenId_, uint256 price_) external {
-        require(property.ownerOf(tokenId_) == msg.sender, "Sender does not own property");
+    function setTaxConfig(ConfigOptions option, uint256 value) external onlyOwner {
+        taxConfig[option] = value;
 
-        _setPrice(tokenId_, price_);
+        emit Config(option, value);
+    }
+
+    // TODO: withraw community treasury
+
+    /**
+     * @dev Set the current price of an Harberger property with token id.
+     *
+     * Emits a {Price} event.
+     */
+    function setPrice(uint256 tokenId, uint256 price) external {
+        if (!property.isApprovedOrOwner(msg.sender, tokenId)) revert Unauthorized();
+        if (price == this.getPrice(tokenId)) return;
+
+        bool success = this.collectTax(tokenId);
+        if (success) _setPrice(tokenId, price);
     }
 
     /**
      * @dev Returns the current price of an Harberger property with token id.
      */
-    function getPrice(uint256 tokenId_) external view returns (uint256 price) {
```

Comment:
so we changed the convention on parameters? no `_` suffix?

Reply:
Hmmm, I thought logbook wasn't using suffix for input args, so I removed them. Looks like I was mistaken.